src/python/pants/util/strutil.py

```diff
@@ -160,14 +160,25 @@ def hard_wrap(s: str, *, indent: int = 0, width: int = 96) -> Sequence[str]:
     ]
 
 
-def bullet_list(elements: Iterable[str]) -> str:
+def bullet_list(elements: Iterable[str], max_elements: int = -1) -> str:
```

Comment:
Probably more clear is using `None`.

Reply:
I considered it, but that makes the implementation less ergonomic, having to cope with the `max_elements` as being a `int | None` rather than just `int`.

Also, the stdlib use this construct quite a bit already, so I think it should be rather familiar concept for any Python developer. 

A couple examples:
```
str.split = split(self, /, sep=None, maxsplit=-1)
[...]
    maxsplit
      Maximum number of splits to do.
      -1 (the default value) means no limit.

str.replace = replace(self, old, new, count=-1, /)
    Return a copy with all occurrences of substring old replaced by new.
    
      count
        Maximum number of occurrences to replace.
        -1 (the default value) means replace all occurrences.

```

Reply:
@Eric-Arellano  Does the thumbs up indicate, ok good as-is, or, yes please change to `None` ? ;)

Reply:
Good as is :)